### PR TITLE
SCP Events

### PR DESCRIPTION
**Pull request by xNexusACS**
TODO LIST:
- [x] Scp173 Events
- [x] Scp939 Events
- [x] Scp049 Events
- [x] Scp0492 Events
- [x] Scp106 Events
- [x] Scp096 Events
- [x] Scp079 Events
- [x] Scp914 Events
- [x] Scp330 Events

### CODE REVIEW

**Comment by xNexusACS** [comment]
Im not going to add the Scp079LockDoor Event since its ability has been removed in 13.0

**Comment by xNexusACS** [comment]
(Note: The Bloodlusting event is executing every second since its made for that, to be executed every second during the bloodlust)

**Comment by xNexusACS** [comment]
Im testing all of the events rn to make sure there are no errors using any property

**Comment by Jesus-QC** [comment]
Will review asap

File: CursedMod/Events/Arguments/SCPs/Scp049/PlayerCallingEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp049;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp049;
+
+public class PlayerCallingEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp049/PlayerCallingEventArgs.cs:15]: Is better to use Scp049UsingCallAbilityEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp049/PlayerFinishReviveEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using CursedMod.Features.Wrappers.Player.Ragdolls;
+using PlayerRoles.PlayableScps.Scp049;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp049;
+
+public class PlayerFinishReviveEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent, ICursedRagdollEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp049/PlayerFinishReviveEventArgs.cs:16]: Scp049ResurrectingPlayerEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp049/PlayerSensingEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp049;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp049;
+
+public class PlayerSensingEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp049/PlayerSensingEventArgs.cs:15]: Scp049UsingSenseAbilityEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp049/PlayerStartReviveEventArgs.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using CursedMod.Features.Wrappers.Player.Ragdolls;
+using PlayerRoles.PlayableScps.Scp049;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp049;
+
+public class PlayerStartReviveEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent, ICursedRagdollEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp049/PlayerStartReviveEventArgs.cs:16]: Scp049StartingResurrectionEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp0492/PlayerBloodlustingEventArgs.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp0492;
+
+public class PlayerBloodlustingEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
+{
+    public PlayerBloodlustingEventArgs(ReferenceHub player, ReferenceHub target)
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp0492/PlayerBloodlustingEventArgs.cs:16]: Scp0492UsingBloodlustAbilityEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp0492/PlayerConsumingCorpseEventArgs.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using CursedMod.Features.Wrappers.Player.Ragdolls;
+using PlayerRoles.PlayableScps.Scp049.Zombies;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp0492;
+
+public class PlayerConsumingCorpseEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent, ICursedRagdollEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp0492/PlayerConsumingCorpseEventArgs.cs:16]: Scp0492ConsumingCorpseEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp0492/PlayerCorpseConsumedEventArgs.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using CursedMod.Features.Wrappers.Player.Ragdolls;
+using PlayerRoles.PlayableScps.Scp049.Zombies;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp0492;
+
+public class PlayerCorpseConsumedEventArgs : EventArgs, ICursedPlayerEvent, ICursedRagdollEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp0492/PlayerCorpseConsumedEventArgs.cs:16]: Scp0492ConsumedCorpseEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerBlackoutRoomEventArgs.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Facility.Rooms;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerBlackoutRoomEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerBlackoutRoomEventArgs.cs:16]: Scp079UsingBlackoutRoomAbilityEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerBlackoutZoneEventArgs.cs [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+using System;
+using CursedMod.Features.Wrappers.Player;
+using MapGeneration;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerBlackoutZoneEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
+{
+    public PlayerBlackoutZoneEventArgs(Scp079BlackoutZoneAbility blackoutZoneAbility, int powerCost, float duration)
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerBlackoutZoneEventArgs.cs:18]: Scp079UsingBlackoutZoneAbilityEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerCancelLockdownEventArgs.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Facility.Rooms;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerCancelLockdownEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerCancelLockdownEventArgs.cs:16]: Scp079CancellingLockdownEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerChangeCameraEventArgs.cs [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp079.Cameras;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerChangeCameraEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerChangeCameraEventArgs.cs:15]: Scp079ChangingCameraEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerChangeDoorLockEventArgs.cs [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+using System;
+using CursedMod.Features.Wrappers.Facility.Doors;
+using CursedMod.Features.Wrappers.Player;
+using Interactables.Interobjects.DoorUtils;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerChangeDoorLockEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerChangeDoorLockEventArgs.cs:17]: Scp079ChangingDoorLockEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerChangeDoorStateEventArgs.cs [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+using System;
+using CursedMod.Features.Wrappers.Facility.Doors;
+using CursedMod.Features.Wrappers.Player;
+using Interactables.Interobjects.DoorUtils;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerChangeDoorStateEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerChangeDoorStateEventArgs.cs:17]: Scp079ChangingDoorStateEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerGainExperienceEventArgs.cs [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerGainExperienceEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerGainExperienceEventArgs.cs:15]: Scp079GainingExperienceEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerLevelUpEventArgs.cs [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerLevelUpEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerLevelUpEventArgs.cs:15]: Scp079LevelingUpEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerMoveElevatorEventArgs.cs [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+using System;
+using CursedMod.Features.Wrappers.Facility.Rooms;
+using CursedMod.Features.Wrappers.Player;
+using Interactables.Interobjects;
+using MapGeneration;
+using PlayerRoles.PlayableScps.Scp079;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerMoveElevatorEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerMoveElevatorEventArgs.cs:18]: Scp079MovingElevatorEventArgs

File: CursedMod/Events/Arguments/SCPs/Scp079/PlayerPingEventArgs.cs [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+// -----------------------------------------------------------------------
+
+using System;
+using CursedMod.Features.Wrappers.Player;
+using PlayerRoles.PlayableScps.Scp079.Pinging;
+using UnityEngine;
+
+namespace CursedMod.Events.Arguments.SCPs.Scp079;
+
+public class PlayerPingEventArgs : EventArgs, ICursedCancellableEvent, ICursedPlayerEvent
```
Comment (Jesus-QC) [inline CursedMod/Events/Arguments/SCPs/Scp079/PlayerPingEventArgs.cs:16]: Scp079UsingPingAbilityEventArgs

**Review by Jesus-QC — APPROVED** [review]
Se ve todo bien, voy a renombrar algunas cosas

**Pull request merged**